tests/test_fields.py

```diff
@@ -630,3 +633,406 @@ def x(self) -> int:
+    page = BigToItemOnlyPage(
+        response=response, select_fields=SelectFields({"x": False})
+    )
+    assert page.fields_to_extract == []
+    assert await page.to_item() == BigItem(x=1, y=2, z=None)
+    with pytest.raises(TypeError, match=expected_type_error_msg):
+        await item_from_select_fields(page)
+
+    # Boolean-like values are not supported. They are simply ignored and the
+    # page would revert back to the default field directives.
+    page = BigToItemOnlyPage(
+        response=response,
+        select_fields=SelectFields({"x": 0, "y": 0, "z": 1}),  # type: ignore[dict-item]
+    )
+    assert page.fields_to_extract == []
+    assert await page.to_item() == BigItem(x=1, y=2, z=None)
+    assert await item_from_select_fields(page) == BigItem(x=1, y=2, z=None)
+
+    # If the item class doesn't have a field, it would error out.
+    fields = {"x": True, "not_existing": True}
+    expected_value_error_msg = (
+        r"The fields {'not_existing'} is not available in <class 'tests."
+        r"test_fields.BigItem'> which has SelectFields\(fields={'x': True, "
+        r"'not_existing': True}\)."
+    )
+    page = BigToItemOnlyPage(response=response, select_fields=SelectFields(fields))
+    with pytest.raises(ValueError, match=expected_value_error_msg):
+        assert page.fields_to_extract
+    assert await page.to_item() == BigItem(x=1, y=2, z=None)
+    with pytest.raises(ValueError, match=expected_value_error_msg):
+        await item_from_select_fields(page)
+
+
+@attrs.define
+class BigUnreliablePage(WebPage[BigItem]):
+    call_counter: DefaultDict = attrs.field(factory=lambda: defaultdict(int))
+
+    @field
+    def x(self):
+        self.call_counter["x"] += 1
+        return 1
+
+    @field(disabled=True)
+    def z(self):
+        self.call_counter["z"] += 1
+        return 3
+
+    async def to_item(self) -> BigItem:
+        return BigItem(x=self.x, y=2, z=self.z)
+
+
+@pytest.mark.asyncio
+async def test_select_fields_but_unreliable() -> None:
+    """This is essentially a combination of ``test_select_fields()`` and
+    ``test_select_fields_but_to_item_only()`` where the ``.to_item()`` method
+    is overridden as well as ``@field`` decorators are partially used.
+
+    For this test, the ``.to_item()`` method is incorrectly made wherein it's
+    not properly checking the ``.fields_to_extract`` to determine fields to
+    populate.
+    """
+    # Required fields from the item cls which are not included raise an TypeError
+    expected_type_error_msg = (
+        r"__init__\(\) missing 1 required positional argument: 'x'"
+    )
+    response = HttpResponse("https://example.com", b"")
+
+    # When SelectFields isn't set
+    page = BigUnreliablePage(response=response)
+    assert page.fields_to_extract == ["x"]
+    assert await page.to_item() == BigItem(x=1, y=2, z=3)
+    assert page.call_counter == {"x": 1, "z": 1}
+    assert await item_from_select_fields(page) == BigItem(x=1, y=None, z=None)
```

Comment:
> If it were to simply call `.to_item()` and drop "z", that _might_ be inefficient since "z" could use additional requests.

> The best route to take here is the users should create page objects with efficient `.to_item()` methods

Since the efficiency of dropping `z` after `to_item()` is a user problem, one that users should address, should we still drop it?

```python
assert await item_from_select_fields(page) == BigItem(x=1, y=2, z=None)
```

*Maybe* also log a warning in scenarios where we drop a field that as not `None`?

Reply:
As discussed last Thursday, we'll need to update `item_from_select_fields()` to always call `.to_item()` and then dropping the fields. 68d2cc2 In this way, the page obj developer needs to make sure that `.to_item()` is efficient (i.e. calling `super().to_item()` or `item_from_fields()` to utilize the selected fields.) Moreover, we'll use `fields_to_ignore` instead of `fields_to_extract`.

Reply:
As discussed last Thursday, we'll need to update \`item\_from\_select\_fields()\` to always call \`.to\_item()\` and then dropping the fields. d09bf66 In this way, the page obj developer needs to make sure that \`.to\_item()\` is efficient (i.e. calling \`super().to\_item()\` or \`item\_from\_fields()\` to utilize the selected fields.) Moreover, we'll use \`fields\_to\_ignore\` instead of \`fields\_to\_extract\`.